pandas/core/indexes/multi.py

```diff
@@ -431,6 +437,70 @@ def from_product(cls, iterables, sortorder=None, names=None):
         codes = cartesian_product(codes)
         return MultiIndex(levels, codes, sortorder=sortorder, names=names)
 
+    @classmethod
+    def from_frame(cls, df, sortorder=None, names=None):
+        """
+        Make a MultiIndex from a DataFrame.
+
+        .. versionadded:: 0.24.0
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            DataFrame to be converted to MultiIndex.
+        sortorder : int or None
+            Level of sortedness (must be lexicographically sorted by that
+            level).
+        names : list-like, optonal
+            If no names are provided, use the column names, or tuple of column
+            names if the columns is a MultiIndex. If a sequence, overwrite
+            names with the given sequence.
+
+        Returns
+        -------
+        MultiIndex or Index
+            The MultiIndex representation of the given DataFrame.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame([['ahc', 'happy'], ['ahc', 'jolly'],
+        ...                    ['boh', 'happy'], ['boh', 'jolly'],
+        ...                    ['oil', 'joy'], ['oil', 'joy']],
+        ...                   columns=['a', 'b'])
+        >>> df
+             a      b
+        0  ahc  happy
+        1  ahc  jolly
+        2  boh  happy
+        3  boh  jolly
+        4  oil    joy
+        5  oil    joy
+
+        >>> pd.MultiIndex.from_frame(df)
+        MultiIndex(levels=[['ahc', 'boh', 'oil'], ['happy', 'jolly', 'joy']],
+                   labels=[[0, 0, 1, 1, 2, 2], [0, 1, 0, 1, 2, 2]],
+                   names=['a', 'b'])
+
+        # Use explicit names, instead of column names
```

Comment:
Better as a paragraph than as a code comment (remove the `#` and leave a blank line after this line.

Reply:
done.